.github/workflows/publish-helm-chart.yml

```diff
@@ -0,0 +1,42 @@
+name: Publish helm chart
+
+concurrency: release-helm
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'main'
+
+jobs:
+  release:
+    # depending on default permission settings for your org (contents being read-only or read-write for workloads), you will have to add permissions
+    # see: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Configure Git
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+
+      - name: Install Helm
+        uses: azure/setup-helm@v3
+        with:
+          version: v3.10.0
+
+      - name: Update chart dependencies
+        run: |
+          helm repo add bitnami https://raw.githubusercontent.com/bitnami/charts/archive-full-index/bitnami
```

Comment:
```suggestion
```

I think as long as you dont reference the chart as bitnami/<chartname> but instead use the full reference, this isn't needed.

Chart.yaml references the helm chart repository by full URL, and not by a pre-added helm chart repo.

Reply:
@consideRatio I thought so too, but it wasn't working for me without this line

Reply:
Hmmm that is weird, because there are no references to `bitnami` as a named `helm repo` anywhere.

It worked for me here [without any `helm repo add`](https://github.com/mastodon/mastodon/pull/20394/files#diff-b4602b45541e364fdb9179d7237a3567d47a71d8166467628cc57628cf6d97fdR24-R41) step.

Reply:
Yeah this is what happens: https://github.com/mastospawner/mastodon/actions/runs/3451321031/jobs/5760465301

```
Error: no repository definition for https://raw.githubusercontent.com/bitnami/charts/archive-full-index/bitnami, https://raw.githubusercontent.com/bitnami/charts/archive-full-index/bitnami, https://raw.githubusercontent.com/bitnami/charts/archive-full-index/bitnami
```

It doesn't make sense to me either

Reply:
😮 

I dont get this, at all, but i suspect that the difference is you providing a positional argument to helm dep. Not the first time im confused by this... Argh

Reply:
`helm install` (which installs exact versions as locked) will definitely require the repo added first. I suspect this is part of the reason the docs recommend `helm dep up` (which pulls latest versions within specified semantic range).

Reply:
Hmmm... I cant reproduce this without involving `cr` (chart-releaser). `helm dep up`, `helm package`, `helm install`. It all works fine even if I've done `rm -rf ~/.helm/cache` before as well `helm repo list` shows no named repositories added.

When inspecting the failure, I note that [it is chart-releaser](https://github.com/helm/chart-releaser-action/blob/98bccfd32b0f76149d188912ac8e45ddd3f8695f/cr.sh#L258-L268) that has a failure doing `helm package` I think.

![image](https://user-images.githubusercontent.com/3837114/201686404-15069f1e-0341-4280-be60-2d6715fbd491.png)

I currently think of this as a workaround for something wrong in chart-releaser.

---

I investigate this more in depth and wrote a summary in https://github.com/helm/chart-releaser-action/issues/74#issuecomment-1313931556.

Practically, this comes down to if we should version control a Chart.lock file or not.

If we should have Chart.lock version controlled, we need to do `helm repo add`, but should not run `helm dep update` but `helm dep build` instead as otherwise the Chart.lock will be updated and practically ignored. 

If we are not to have Chart.lock version controlled, we can make this workflow job run properly as long as there isn't a Chart.lock around, which can happen as long as `helm dep update` isn't called before the chart-releaser-action runs, as that emits a Chart.lock.

For now, my suggestion for now is to: remove Chart.lock, and remove `helm repo add` + `helm dep up` calls before chart-releaser-action is run.

Reply:
@consideRatio I'd prefer to keep `Chart.lock` and use `helm dep build` instead. From what I've seen, it's best practice to version control the `Chart.lock` file to ensure the exact versions of dependencies are installed.

Reply:
I guess, but also a bit more work.

We need to ensure that users version bumping in Chart.yaml re-freezes the Chart.lock file then as well.

I'm happy to see that `helm dep build` will realize if Chart.yaml and Chart.lock is out of sync, so if we go for that, we should update our test suite to also do `helm dep build` instead of `helm dep update`. They will then also due to this need to have `helm repo add whatever-name <bitnami repo url>`.

```
helm dep build
Error: the lock file (Chart.lock) is out of sync with the dependencies file (Chart.yaml). Please update the dependencies
```